### PR TITLE
remove substation and voltage level

### PR DESCRIPTION
**Pull request by etiennehomer**
Signed-off-by: Etienne Homer <etiennehomer@gmail.com>

### CODE REVIEW

File: src/test/java/org/gridsuite/modification/server/ModificationControllerTest.java [hunk 1]
```diff
@@ -1125,6 +1125,28 @@ public void testDeleteEquipment() {
...
+        webTestClient.delete().uri(uriString, TEST_NETWORK_ID, "SUBSTATION", "s3")
+                .exchange()
+                .expectStatus().isOk()
+                .expectHeader().contentType(MediaType.APPLICATION_JSON)
+                .expectBodyList(EquipmentDeletionInfos.class)
+                .value(modifications -> modifications.get(0),
+                        MatcherEquipmentDeletionInfos.createMatcherEquipmentDeletionInfos(ModificationType.EQUIPMENT_DELETION, "s3", "SUBSTATION", Set.of()));
+
+        testNetworkModificationsCount(TEST_GROUP_ID, 15);
     }
```
Comment (FranckLecuyer) [inline src/test/java/org/gridsuite/modification/server/ModificationControllerTest.java:1150]: Maybe test also failing voltage level and substation deletion (when lines or transformers are present)
Test also that all equipments in the voltage level are properly deleted (injections ...)

**Review by FranckLecuyer — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/modification/server/service/NetworkStoreListener.java [hunk 2]
```diff
@@ -210,6 +210,8 @@ public void storeSubstationCreation(SubstationCreationInfos substationCreationIn
         } else if (identifiable instanceof HvdcLine) {
             ids.add(((HvdcLine) identifiable).getConverterStation1().getTerminal().getVoltageLevel().getSubstation().orElseThrow().getId()); // TODO
             ids.add(((HvdcLine) identifiable).getConverterStation2().getTerminal().getVoltageLevel().getSubstation().orElseThrow().getId()); // TODO
+        } else if (identifiable instanceof VoltageLevel) {
+            ids.add(((VoltageLevel) identifiable).getSubstation().orElseThrow().getId());
         }
```
[the remark this answers is not in the document]
Comment (etiennehomer) [reply src/main/java/org/gridsuite/modification/server/service/NetworkStoreListener.java:215]: The id of the deleted substation is set in deletedEquipmentId. We don't need to set it in the substations to be updated.
For the adjacent substations, it should be done in this dev or later if we change the behavior of substation.remove() ?
@geofjamg @SlimaneAmar

File: src/test/java/org/gridsuite/modification/server/ModificationControllerTest.java (hunk 1 above)
Comment (etiennehomer) [reply src/test/java/org/gridsuite/modification/server/ModificationControllerTest.java:1167]: I added tests when deletion fails and some check that the substation or the vl is deleted. 
For injections that are deleted, this behavior is tested in NetworkStoreIt

**Review by JacquesBorsenberger — APPROVED** [review]

File: src/main/java/org/gridsuite/modification/server/service/NetworkStoreListener.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (etiennehomer) [reply src/main/java/org/gridsuite/modification/server/service/NetworkStoreListener.java:215]: Discussed with Geoffroy: we'll do the evolution if needed later for the adjacent substations update . For performances, it's better to wait before doing it (no useless update). But we have to keep in mind that there will be something to change here

**Review by FranckLecuyer — APPROVED** [review]

**Pull request merged**